android-interop-testing spews lots of warnings

For the generated code, we should probably disable the warning, like we do elsewhere:
https://github.com/grpc/grpc-java/blob/9dec06cc91b5ff6e70771994963b87f561bb6f9b/protobuf-lite/build.gradle#L29-L32

The rest look like they'll need changes, but I just too a cursory glance.

```
> Task :grpc-android-interop-testing:compileDebugJavaWithJavac
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/EmptyProtos.java:112: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Messages.java:286: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Messages.java:567: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Messages.java:891: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Messages.java:1715: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Messages.java:2908: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Messages.java:3584: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Messages.java:4011: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Messages.java:4411: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Messages.java:4848: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Messages.java:5548: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Messages.java:6120: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Messages.java:6415: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Messages.java:6757: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Messages.java:7141: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Messages.java:7621: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Metrics.java:354: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Metrics.java:774: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/build/generated/source/proto/debug/java/io/grpc/testing/integration/Metrics.java:1010: warning: [cast] redundant cast to Builder
      return (Builder) DEFAULT_INSTANCE.createBuilder(prototype);
             ^
grpc-java/android-interop-testing/src/main/java/io/grpc/android/integrationtest/InteropTask.java:72: warning: [Finally] If you return or throw from a finally, then values returned or thrown from the try-catch block will be ignored. Consider using try-with-resources instead.
        throw ex;
        ^
    (see https://errorprone.info/bugpattern/Finally)
grpc-java/android-interop-testing/src/main/java/io/grpc/android/integrationtest/InteropTask.java:74: warning: [Finally] If you return or throw from a finally, then values returned or thrown from the try-catch block will be ignored. Consider using try-with-resources instead.
        throw new RuntimeException(ex);
        ^
    (see https://errorprone.info/bugpattern/Finally)
grpc-java/interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java:294: warning: [UnnecessaryAnonymousClass] Implementing a functional interface is unnecessary; prefer to implement the functional interface method directly and use a method reference instead.
  private final ClientInterceptor tracerSetupInterceptor = new ClientInterceptor() {
                                  ^
    (see https://errorprone.info/bugpattern/UnnecessaryAnonymousClass)
  Did you mean 'private  <ReqT, RespT> ClientCall<ReqT, RespT> tracerSetupInterceptor('?
22 warnings
```

--- a/android-interop-testing/build.gradle
+++ b/android-interop-testing/build.gradle
@@ -87,4 +87,14 @@ task checkStyleTest(type: Checkstyle) {
 
 project.tasks['check'].dependsOn checkStyleMain, checkStyleTest
 
+import net.ltgt.gradle.errorprone.CheckSeverity
+
+tasks.withType(JavaCompile) {
+    options.compilerArgs += [
+            "-Xlint:-cast"
+    ]
+    // Reuses source code from grpc-interop-testing, which targets Java 7 (no method references)
+    options.errorprone.check("UnnecessaryAnonymousClass", CheckSeverity.OFF)
+}
+
 configureProtoCompilation()

--- a/android-interop-testing/src/main/java/io/grpc/android/integrationtest/InteropTask.java
+++ b/android-interop-testing/src/main/java/io/grpc/android/integrationtest/InteropTask.java
@@ -53,6 +53,7 @@ final class InteropTask extends AsyncTask<Void, Void, String> {
     tester.setUp();
   }
 
+  @SuppressWarnings("Finally")
   @Override
   protected String doInBackground(Void... ignored) {
     try {

